### PR TITLE
fixed - error 500 when the delete button is clicked

### LINKED ISSUE #5240: Edit page - delete button error
### Package

filament/filament

### Package Version

v2.16.55

### Laravel Version

v9

### Livewire Version

v2

### PHP Version

8.1

### Problem description

Edit page - Error 500 when the delete button is clicked.
https://www.veed.io/view/88817495-3db8-4d58-8e71-a6b0e2e1b2b0?sharingWidget=true&panel=share

### Expected behavior

200 status code

### Steps to reproduce

Go to the posts page and edit a post. Ex: https://demo.filamentphp.com/blog/posts/9/edit?activeRelationManager=0
Click on the delete button.

### Reproduction repository

https://github.com/leandrocfe/filament-demo

### Relevant log output

```shell
local.ERROR: Missing required parameter for [Route: filament.resources.posts.view] [URI: admin/posts/{record}] [Missing parameter: record]...
```

### PR DESCRIPTION
**Pull request by leandrocfe**
fixed - error 500 when the delete button is clicked
issue [#5240](https://github.com/filamentphp/filament/issues/5240)

### CODE REVIEW

**Comment by danharrin** [comment]
Hey! Could you please explain how this fixes?

**Comment by leandrocfe** [comment]
> Hey! Could you please explain how this fixes?

Calling $this->halt() in the end of the action method. So, the error disappeared

File: packages/admin/src/Pages/Actions/DeleteAction.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ protected function setUp(): void
         $this->keyBindings(['mod+d']);
 
         $this->hidden(static function (Model $record): bool {
-            if (! method_exists($record, 'trashed')) {
+            if (!method_exists($record, 'trashed')) {
```
Comment (zepfietje) [inline packages/admin/src/Pages/Actions/DeleteAction.php]:
```suggestion
            if (! method_exists($record, 'trashed')) {
```

File: packages/admin/src/Pages/Actions/DeleteAction.php [hunk 2]
```diff
@@ -45,13 +45,14 @@ protected function setUp(): void
         $this->action(function (): void {
             $result = $this->process(static fn (Model $record) => $record->delete());
 
-            if (! $result) {
+            if (!$result) {
```
Comment (zepfietje) [inline packages/admin/src/Pages/Actions/DeleteAction.php]:
```suggestion
            if (! $result) {
```

**Comment by danharrin** [comment]
While it worked, it wasn't a perfect solution. But it did allow me to discover a fix we can make on our end that works in all cases.

It is still a Livewire bug though, we shouldn't have to do it like this.

**Comment by leandrocfe** [comment]
> While it worked, it wasn't a perfect solution. But it did allow me to discover a fix we can make on our end that works in all cases.
> 
> It is still a Livewire bug though, we shouldn't have to do it like this.

Great! I'm not able to fix it as well as you can 😅

**Comment by danharrin** [comment]
I wouldn't have even tried halting... that was the key to me discovering one of the causes of this bug

**Comment by danharrin** [comment]
So thank you :)

**Comment by zepfietje** [comment]
Nice work guys, @leandrocfe @danharrin!

**Comment by diegobas** [comment]
Is it possible that this error has already been fixed by Livewire? See: https://github.com/livewire/livewire/pull/5574

Anyway, thank you for your work!

**Comment by zepfietje** [comment]
Seems unrelated to me, @diegobas.

**Pull request merged**